python/cugraph/structure/number_map.py

```diff
@@ -0,0 +1,263 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cugraph
+
+class NumberMap:
+    class SingleGPU:
+        def __init__(self, df, col_names):
+            # TODO: Fill in implementation
+            pass
+
+        def append(self, df, col_names):
+            # TODO: Fill in implementation
+            pass
+
+        def compute(self):
+            # TODO: Fill in implementation
+            pass
+
+        def to_vertex_id(df, col_names):
+            # TODO: Fill in implementation
+            pass
+
+        def from_vertex_id(series):
+            # TODO: Fill in implementation
+            pass
+
+    class MultiGPU:
+        def __init__(self, ddf, col_names):
+            # TODO: Fill in implementation
+            pass
+
+        def append(self, ddf, col_names):
+            # TODO: Fill in implementation
+            pass
+
+        def compute(self):
+            # TODO: Fill in implementation
+            pass
+
+        def to_vertex_id():
+            # TODO: Fill in implementation
+            pass
+
+        def from_vertex_id():
+            # TODO: Fill in implementation
+            pass
+
+            
+    def __init__(self):
+        self.implementation = None
+
+    def compute_vals(column_names):
+        """
+        Helper function to compute internal column names based on external column names
+        """
+        return [str(i) for i in range(len(column_names))]
+
+    def from_dataframe(self, df, col_names):
```

Comment:
This PR can't close #813 without integration into 0.15 OPG pipeline (#944, #947, #964) and renumbering tests passing.

Reply:
Added more details on this in my latest review. Resolving this one.